include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -863,26 +872,19 @@ SpatialOperatorBase<dim, Number>::get_constraint_u() const
   return constraint_u;
 }
 
-template<int dim, typename Number>
-double
-SpatialOperatorBase<dim, Number>::get_viscosity() const
-{
-  return param.viscosity;
-}
-
 template<int dim, typename Number>
 dealii::VectorizedArray<Number>
 SpatialOperatorBase<dim, Number>::get_viscosity_boundary_face(unsigned int const face,
                                                               unsigned int const q) const
 {
-  dealii::VectorizedArray<Number> viscosity =
-    dealii::make_vectorized_array<Number>(get_viscosity());
-
-  bool const viscosity_is_variable = param.use_turbulence_model;
-  if(viscosity_is_variable)
-    viscous_kernel->get_coefficient_face(face, q);
-
-  return viscosity;
+  if(param.viscosity_is_variable())
+  {
+    return viscous_kernel->get_coefficient_face(face, q);
+  }
+  else
+  {
+    return dealii::make_vectorized_array<Number>(param.viscosity);
+  }
```

Comment:
I think this could be a useful general optimization for the viscous kernel (i.e. shifting this if-statement inside the viscous kernel for all getter functions). In case the viscosity is constant and we access the viscosity via `viscous_kernel->get_...()`, we should only cause data transfer from main memory if really needed.  @kronbichler What do you think? (there might be applications where we do not just read boundary face data, but for all q-points (cells, faces, boundary faces) and where performance becomes more relevant.) The downside could be another if-statement to be evaluated in viscous kernel even in cases where we did not evaluate such an if-statement before (like in the current implementation when accessing cell-data).

Of course, this is not directly related to the present PR and could be done as follow-up work.

Reply:
I think I will not change this for now to not touch the `viscous_kernel` as well in this PR, once we have it in `viscous_kernel`, we can skip it here ... alright?

Reply:
I created issue #413 and resolve this conversation now.